### PR TITLE
add chapter 01, 02, 04  example code, and chapter 05 part

### PR DESCRIPTION
**Pull request by JianmingXia**
code licensed as MIT;
code in 'node-async' sub dir.

### CODE REVIEW

File: node/node-async/yarn.lock [hunk 1]
```diff
@@ -0,0 +1,1045 @@
...
+
+wrappy@1:
+  version "1.0.2"
+  resolved "https://registry.yarnpkg.com/wrappy/-/wrappy-1.0.2.tgz#b5243d8f3ec1aa35f1364605bc0d1036e30ab69f"
+
+write@^0.2.1:
+  version "0.2.1"
+  resolved "https://registry.yarnpkg.com/write/-/write-0.2.1.tgz#5fc03828e264cea3fe91455476f7a3c566cb0757"
+  dependencies:
+    mkdirp "^0.5.1"
```
Comment (josiahcarlson) [inline node/node-async/yarn.lock]: Am I correct, is this like 50+ dependencies just to run this? Is this all of your packages installed on your system? Please clean this up a bit.

Comment (JianmingXia) [reply node/node-async/yarn.lock]: Dependencies config are in package.json actually. 
```
    "dependencies": {
        "ioredis": "^3.2.2",
        "uuid": "^3.3.2"
    },
    "devDependencies": {
        "@types/ioredis": "^3.2.14",
        "eslint": "^5.3.0",
        "mocha": "^5.2.0",
        "should": "^13.2.3"
    }
```

In order to get consistent installs across machines so I add yarn.lock file.  You can see the document in https://yarnpkg.com/en/docs/yarn-lock.

Comment (josiahcarlson) [reply node/node-async/yarn.lock]: Just to make sure I got this, 2x dependencies to run, 4x to develop, but the full dependency tree from those 4x dev dependencies turns into 185 total packages (184 if you exclude Node.js itself). Wow. That sounds painful.

Okay, snyk.io doesn't show any current vulnerabilities for the direct dependencies (`ioredis`, `eslint`, `mocha`, `should`), but I have no idea about the rest of the files in the the yarn.lock file. Have you checked `npm audit` output for any / all of the packages in yarn.lock (I don't want to accidentally introduce a vulnerability to a future reader)?

Comment (JianmingXia) [reply node/node-async/yarn.lock]: I have not find ```audit command``` in yarn, so I change install way.
Now I run ```npm install``` command and generate a **package-lock.josn** file. And use **npm audit**：
```
=== npm audit security report ===

found 0 vulnerabilities
 in 295 scanned packages
```

I'm sorry I have no attention in dependencies vulnerability. Do you think so?

Comment (josiahcarlson) [reply node/node-async/yarn.lock]: Thank you for running the dependency vulnerability check. I just wanted to make sure that we are not pointing to some (sub-)dependency that has known vulnerabilities.

Your changes look good. Thank you, merging momentarily.

**Pull request merged**